solidity/contracts/converter/BancorConverterRegistry.sol

```diff
@@ -315,8 +415,15 @@ contract BancorConverterRegistry is IBancorConverterRegistry, ContractRegistryCl
         // verify that the converter holds balance in each of its reserves
         uint reserveTokenCount = _converter.connectorTokenCount();
         for (uint i = 0; i < reserveTokenCount; i++) {
-            if (_converter.connectorTokens(i).balanceOf(_converter) == 0)
-                return false;
+            IERC20Token reserveToken = _converter.connectorTokens(i);
+            if (reserveToken != IERC20Token(0)) {
+                if (reserveToken.balanceOf(_converter) == 0)

```

Comment:
the new `reserveBalance(address)` can be used here instead (I might have done this already in 0.6.0)

Reply:
@yudilevi : Of course. I left it as is because I was not sure about your final version of this function. This will also resolve the eth-reserve address problem here.

Reply:
cool

Reply:
@yudilevi : I created this branch off of 0.5.19 (or maybe `remove_ether_token`, but most certainly not from 0.6.0). So we can do this only after merging this branch to 0.6.0.
I suggest opening a new task for this merge, and then add all of the required adjustments (mentioned in some of your comments above) in a checklist.

Reply:
pulling the target branch into this branch doesn't require a new task, it's part of a normal task workflow - it can't be merged anyway before doing that.

Reply:
@yudilevi : after merging this branch to 0.6.0, we need to remember to replace the balance checking code with calling the `reserveBalances` function. We also need to use the `protected` modifier mentioned in some of your comments above.
As far as I understand:
1. we cannot do this before merging
2. we can forget doing this after merging
That's why I suggested to list these details explicitly in a Trello task.

Reply:
sure - we can definitely add a task for these additions

Reply:
@yudilevi : please see updated code; this was fixed in the (now merged) `liquidity_balances` branch.